test/linalg.jl

```diff
@@ -52,6 +52,7 @@ end
     @test loglikelihood(fit!(wm1)) ≈ loglikelihood(m1)
 end
 
+#=  I don't see this testset as meaningful b/c diagonal A does not occur after amalgamation of ReMat's for the same grouping factor - D.B.
```

Comment:
I'm trying to think of a case where diagonal A could occur. Perhaps in one of the custom model types some of the ZiF spinoff projects are working on? Maybe this is something re-visit in a few weeks to months and see if we can further trim.

Reply:
It is best to think of how `rankUpdate!` gets called.  After pre- and post-multiplication by Lambda and inflation of the diagonal the `rankUpdate!` methods get called with `C` being one of the diagonal blocks and `A` being a block to the left of `C`.  So `C` is `m.L[Block(2,2)]` or higher because there are no blocks to the left of `m.L[Block(1,1)]`.  The cases that matter are when there is more than one grouping factor for the random effects.  If `A` were to be diagonal then it would need to be generated from two scalar random-effects terms, as in `(1|G) + (1|H)` and the number of levels of `G` and `H` would need to be equal and the levels would match.  You can get that from `(1|G) + (0+x|G)` but that is now collapsed to a single r.e. term either through `zerocorr(1+x|G)` or by amalgamation of the random effects terms.  So I don't think there will be real-life cases where `A` is diagonal.  `C` is always a square matrix and either `Diagonal` or `UniformBlockDiagonal` or, through fill-in, `Dense`.   The only special cases for `A` are `Dense`, `SparseMatrixCSC` or, in a new branch that I haven't pushed yet, `KHotColumn`.  The last one is a special type of sparse matrix generated by nested grouping factors for vector-valued random effects where each column of `A` has exactly `K` adjacent non-zeros.

Reply:
Right, so the only way that this could occur would be if someone had two identical grouping variables so that they wouldn't be amalgamated. Until we see a legitimate use case for that, I think we can safely ignore that possibility.

Thanks for the refresher! :smile: